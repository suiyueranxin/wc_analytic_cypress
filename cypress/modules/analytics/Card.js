class Card {
    constructor(cardIndex) {
        this.cardIndex = cardIndex;
        this.rootPath = 'div.card:eq('+this.cardIndex+')';
    }
    
    /********************************************************************
     * check points
    ********************************************************************/
    checkCardHeader() {
        cy.get(this.rootPath + ' .card-header span.card-title').toMatchDomTextSnapShot();
        cy.get(this.rootPath + ' .card-header span.card-subtitle').toMatchDomTextSnapShot()
    }
    checkKPILineCard() {
        cy.get(this.rootPath + ' .kpiNumber .sapMNCValueScr').then($value => {
            cy.wrap($value).toMatchDomTextSnapShot();
            if ($value.text() != '0') {
                cy.get(this.rootPath + ' .kpiNumber .sapMNCScale').toMatchDomTextSnapShot();
            }
        });
    }
    
    checkLineCard() {

    }

    checkStackedColumnCard() {

    }

    checkTableCard() {
        cy.get(this.rootPath + ' .counter').toMatchDomTextSnapShot();
    }

    checkDonutCard() {

    }

    checkHeatmapCard() {

    }

    checkTotal() {
        cy.get(this.rootPath + ' span[id$="dataset_sum"]').toMatchDomTextSnapShot();
        cy.get(this.rootPath + ' div[id*="details"] span.sapMBtnContent').toMatchDomTextSnapShot();
    }

    checkDetails() {
        cy.get('div[id*=details] .sapMLIBContent:eq(0)').toMatchDomTextSnapShot();
    }
    
    /********************************************************************
     * get element content
    ********************************************************************/
 

    /********************************************************************
     * change element content
    ********************************************************************/
    selectDataPoints(indexArr) {
        if (typeof indexArr == 'number') {
            indexArr = [indexArr];
        }
        indexArr.forEach(index => {
            cy.get(this.rootPath + ' g[data-id="' + index + '"]').first().click({force: true});
        })
    }

    showDetails() {
        cy.get(this.rootPath + ' div[id*=details] button').click({force: true});
    }

    hoverDataPoint(index) {
        cy.get(this.rootPath + ' g.v-datapoint:eq(' + index + ')').hover();
    }

    selectLegends(indexArr) {
        if (typeof indexArr == 'number') {
            indexArr = [indexArr];
        }
        indexArr.forEach(index => {
            cy.get(this.rootPath + ' div.card:eq(1) g.v-legend-item').click({force: true});
        })
    }

    clickHeader() {
        cy.get('div.card-header:eq('+ this.cardIndex + ')').click();
    }
}
  
export default Card;