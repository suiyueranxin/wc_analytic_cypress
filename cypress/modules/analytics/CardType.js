export const KPILINE = 'line';
export const COLUMN = "column";
export const BAR = "bar";
export const STACKED_COLUMN = "stacked column";
export const STACKED_BAR = "stacked bar";
export const PIE = "pie";
export const DONUT = "donut";
export const HEATMAP = "heatmap";
export const TABLE = "table";