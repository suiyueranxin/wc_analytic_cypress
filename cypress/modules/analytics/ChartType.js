export const LINE = 'Line';
export const COLUMN = "Column";
export const BAR = "Bar";
export const STACKED_COLUMN = "Stacked Column";
export const STACKED_BAR = "Stacked Bar";
export const PIE = "Pie";
export const DONUT = "Donut";
export const HEATMAP = "Heat Map";
export const TABLE = "Table";