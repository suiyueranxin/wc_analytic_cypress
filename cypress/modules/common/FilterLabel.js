export const POSTING_DATE = 'Posting Date';
export const DOCUMENT_TYPE_DISPLAY_NAME = 'Document Type Display Name';
export const BUSINESS_PARTNER_CODE = 'Business Partner Code';
export const BUSINESS_PARTNER_GROUP_NAME = 'Business Partner Group Name';