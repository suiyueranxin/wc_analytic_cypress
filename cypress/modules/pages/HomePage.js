import BasePage from './BasePage'

class HomePage extends BasePage {
    constructor() {
        super("/webx/index.html#Shell-home")
    }

    checkEmptyGroup(group) {
        cy.get('.sapUshellTileContainerContent:has(h2[aria-label*="'+group+'"]) ul li[aria-label^="Empty group"]');
    }

    openTile(name) {
        cy.get('.sapUshellTileWrapper:has(span[id$="title-inner"]:contains("'+name+'"))').last().click();
    }

    removeTile(name) {
        cy.get('.sapUshellTileWrapper:has(span[id$="title-inner"]:contains("'+name+'")) div.sapUshellTileDeleteClickArea').last().click();
    }
}

export default HomePage;