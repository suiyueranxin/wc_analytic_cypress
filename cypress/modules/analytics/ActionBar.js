import * as ChartType from '../analytics/ChartType'

class ActionBar {
    constructor() {
        this.rootPath = 'div.sapSuiteUiCommonsChartContainerToolBarArea';
    }
    
    /********************************************************************
     * check points
    ********************************************************************/
   checkTotal(chartType) {
        if (chartType != ChartType.TABLE) {
            cy.get(this.rootPath + ' span[id$="dataset_sum"]').toMatchDomTextSnapShot();
            cy.get(this.rootPath + ' div[id*="details"] span.sapMBtnContent').toMatchDomTextSnapShot();
        } else {
            cy.get(this.rootPath + ' span[id$="dataset_sum"]').should('not.be.visible');
            cy.get(this.rootPath + ' div[id*="details"] span.sapMBtnContent').should('not.be.visible')
        }
    }

    checkDetails() {
        cy.get('div[id*=details] .sapMLIBContent:eq(0)').toMatchDomTextSnapShot();
    }

    /********************************************************************
     * get element content
    ********************************************************************/


    /********************************************************************
     * change element content
    ********************************************************************/
    enterExitFullScreen() {
        cy.get('button[title*="Full Screen"]').click({force: true});
    }

    zoomIn() {
        cy.get('button[title="Zoom In"]').click({force: true});
    }

    zoomOut() {
        cy.get('button[title="Zoom Out"]').click({force: true});
    }

    showHideLegend() {
        cy.get('button[title="Legend"]').click({force: true});
    }

    showHideDetails() {
        cy.get('button:has(span:contains("Details"))').click({force: true});
    }

    switchChartType(chartType) {
        cy.get('li[title^="' + chartType + '"]').click({force: true});
        cy.wait(1000);
    }

}
  
export default ActionBar;