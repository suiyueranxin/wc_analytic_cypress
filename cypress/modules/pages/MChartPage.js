import AnalyticPage from './AnalyticPage'
import ActionBar from '../analytics/ActionBar';
import BindingSelector from '../analytics/BindingSelector';
import Chart from '../analytics/Chart';
import { Log } from '../util/Log';

class MChartPage extends AnalyticPage {
    constructor(viewId) {
        super("/webx/index.html#webclient-ANLY&/Objects/ANLY/ChartContainer?view="+viewId)
        this.actionBar = new ActionBar();
        this.bindingSelector = new BindingSelector();
        this.chart = new Chart();
    }

    /********************************************************************
     * check points
    ********************************************************************/
    @Log()
    checkChartTypeSwitch(switchFlow) {
        switchFlow.forEach(chartType => {
            this.actionBar.switchChartType(chartType);
            this.bindingSelector.checkAllBindings(chartType);
        });
    }

    @Log()
    checkOverAll() {
        this.variant.checkCurrentVariant();
        this.filter.checkCurrentFilters();
        this.chart.checkChartType();
        this.bindingSelector.checkAllBindings();    
    }

    @Log()
    checkWarning(chartType, show) {
        this.bindingSelector.checkWarningBox(chartType, show);
        this.chart.checkWarningMessage(show)
    }

    @Log('Image snapshot of chart')
    checkChartDisplay() {
        this.chart.checkChartDisplay();
    }

    /********************************************************************
     * get element content
     ********************************************************************/


    /********************************************************************
     * change element content
     ********************************************************************/

}

export default MChartPage;