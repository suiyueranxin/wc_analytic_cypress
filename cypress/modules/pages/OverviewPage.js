import AnalyticPage from './AnalyticPage'
import Card from '../analytics/Card'
import * as CardType from '../analytics/CardType'
import * as FilterLabel from '../common/FilterLabel'
import * as Global from '../common/Global'

class OverviewPage extends AnalyticPage {
    constructor() {
        super("/webx/index.html#webclient-anly&/Objects/ANLY/Dashboard")
    }
    /********************************************************************
     * check points
    ********************************************************************/
    checkCards() {
      cy.toMatchImageSnapshot();
    }
    
    checkCard(cardIndex, cardType) {
      let card = new Card(cardIndex);
      card.checkCardHeader();
      if (cardType == CardType.KPILINE) {
        card.checkKPILineCard();
      } else if (cardType == CardType.TABLE) {
        card.checkTableCard();
      }
    }

    checkOverall() {
      this.variant.checkCurrentVariant();
      this.filter.checkCurrentFilters();
      this.checkKPI(0);
    }

    checkKPI(cardIndex) {
      let card = new Card(cardIndex);
      card.checkKPILineCard();
    }

    /********************************************************************
     * get element content
    ********************************************************************/


    /********************************************************************
     * change element content
    ********************************************************************/
    naviageToMchart(cardIndex) {
      let card = new Card(cardIndex);
      card.clickHeader();
      cy.wait(Global.WAIT);
    }

    changeDataByFilter() {
      this.filter.selectDateFilter(FilterLabel.POSTING_DATE, ['01/01/2016...03/31/2016'], true);
    }
}

export default OverviewPage;