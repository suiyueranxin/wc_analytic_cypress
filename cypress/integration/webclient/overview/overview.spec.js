/// <reference types="Cypress" />

import OverviewPage from "../../../modules/pages/OverviewPage"
import * as FilterLabel from "../../../modules/common/FilterLabel"
import MChartPage from "../../../modules/pages/MChartPage";
import * as ChartType from "../../../modules/analytics/ChartType";
import Card from "../../../modules/analytics/Card";
import * as CardType from "../../../modules/analytics/CardType";
import HomePage from "../../../modules/pages/HomePage"

describe('Sales Analysis Overview', () => {
    before(() => {
    });

    beforeEach(() => {
        cy.fixture('user').then(user => {
            cy.login(user.super)
        });
    });

    afterEach(() => {
        let overview = new OverviewPage();
        overview.variant.removeAll();
    });
  
    it('Open page at first time', () => {
        cy.total(9);

        cy.step(1, 'open overview page')
        let overview = new OverviewPage();
        overview.visit();
        
        cy.step(2, 'check variant')
        overview.variant.checkCurrentVariant();
        
        cy.step(3, 'check filter settings')
        overview.filter.checkCurrentFilters();

        cy.step(4, 'check kpi line card display')
        overview.checkCard(0, CardType.KPILINE);

        cy.step(5, 'check line card display')
        overview.checkCard(1, CardType.LINE);
        
        cy.step(6, 'check stacked column card display')
        overview.checkCard(2, CardType.STACKED_COLUMN);

        cy.step(7, 'check table card display')
        overview.checkCard(3, CardType.TABLE);

        cy.step(8, 'check donut card display')
        overview.checkCard(4, CardType.DONUT);

        cy.step(9, 'check heatmap card display')
        overview.checkCard(5, CardType.HEATMAP);          
    });

    it('Change filters of different data types and operators', () => {
        cy.total(5);

        cy.step(1, 'open overview page')
        let overview = new OverviewPage();
        overview.visit();

        cy.step(2, 'set date type filter: Posting Date')
        overview.filter.selectDateFilter(FilterLabel.POSTING_DATE, ['01/01/2016...03/31/2016'], true);
        overview.checkKPI(0);

        cy.step(3, 'set enum type filter: Document Type Display Name')
        overview.filter.selectEnumFilter(FilterLabel.DOCUMENT_TYPE_DISPLAY_NAME, ['A/R Invoice', 'Sales Order'], true);
        overview.checkKPI(0);

        cy.step(4, 'set string type filter (Code): Business Partner Code')
        overview.filter.selectFilterFromList(FilterLabel.BUSINESS_PARTNER_CODE, ['C25000','C60000'], true);
        overview.checkKPI(0);

        cy.step(5, 'set string type filter (Name): Business Parnter Group Name')
        overview.filter.selectFilterFromList(FilterLabel.BUSINESS_PARTNER_GROUP_NAME, ['Customers', 'High Tech'], true);
        overview.checkKPI(0);

        //operator: Equal To

        //operator: Contains

        //operator: Starts with

        //operator: Ends with
    });

    it.skip('Show details', () => {
        cy.total(7);

        cy.step(1, 'open overview page')
        let overview = new OverviewPage();
        overview.visit();

        cy.step(2, 'Change one filter to make cards have data')
        overview.changeDataByFilter();

        [0, 1, 2, 4, 5].forEach((cardIndex, index) => {
            let card = new Card(cardIndex);
            cy.step(index+3+'-1', 'select a datapoint and check sum total')
            //TODO: Details will be from enable to disabled after click a data point
            card.selectDataPoints(0);
            card.checkTotal();

            cy.step(index+3+'-2', 'open details and check details content')
            card.showDetails();
            card.checkDetails();

            if (index == 1) { // no total
                cy.step(index+3+'-3', 'for dual line chart, select another datapoint and check no sum total')
                card.selectDataPoints(3);
                card.checkTotal();

                cy.step(index+3+'-4', 'for dual line chart, open details and check details content')
                card.showDetails();
                card.checkDetails();
            }
        })
    });

    it('Navigate to chart', () => {
        let overview = new OverviewPage();
        overview.visit();
        overview.changeDataByFilter();
        
        let charts = [
            ChartType.LINE,
            ChartType.LINE,
            ChartType.STACKED_COLUMN,
            ChartType.TABLE,
            ChartType.DONUT,
            ChartType.HEATMAP
        ]
        charts.forEach((chartType, index) => {
            overview.naviageToMchart(index);
            let mchart = new MChartPage(chartType);
            mchart.chart.checkChartType(chartType);
            mchart.bindingSelector.checkAllBindings(chartType);
            mchart.back();
        })
        
    });

    it('Save as, update, remove and switch variant', () => {
        let overview = new OverviewPage();
        overview.visit();

        overview.filter.selectDateFilter(FilterLabel.POSTING_DATE, ['01/01/2016...03/31/2016'], true);
        overview.variant.saveAs('AT variant 1');

        overview.revisit();
        
        overview.variant.switchTo('AT variant 1');
        overview.variant.checkCurrentVariant();
        overview.filter.checkCurrentFilters();
        overview.checkKPI(0);

        overview.filter.selectEnumFilter(FilterLabel.DOCUMENT_TYPE_DISPLAY_NAME, ['A/R Invoice', 'Sales Order'], true);
        overview.variant.save();

        overview.variant.switchTo('Standard');
        overview.variant.checkCurrentVariant();

        overview.variant.switchTo('AT variant 1')
        overview.variant.checkCurrentVariant();
        overview.checkKPI(0);   
    });

    it('Set default variant', () => {
        cy.total(9);

        cy.step(1, 'open overview page')
        let overview = new OverviewPage();
        overview.visit();

        cy.step(2, 'change filter and save as variant 1 with default variant')
        overview.filter.selectDateFilter(FilterLabel.POSTING_DATE, ['01/01/2016...03/31/2016'], true);
        overview.variant.saveAs('variant 1', true, true);

        cy.step(3, 'change filter and save as variant 2')
        overview.filter.selectEnumFilter(FilterLabel.DOCUMENT_TYPE_DISPLAY_NAME, ['A/R Invoice', 'Sales Order'], true);
        overview.variant.saveAs('variant 2');

        cy.step(4, 'check default variant is "variant 1"')
        overview.revisit();
        overview.variant.checkCurrentVariant();

        cy.step(5, 'set variant 2 as default variant and check')
        overview.variant.setAsDefault('variant 2');
        overview.revisit();
        overview.variant.checkCurrentVariant();

        cy.step(6, 'set standard as default variant and check')
        overview.variant.setAsDefault('Standard');
        overview.revisit();
        overview.variant.checkCurrentVariant();
 
        cy.step(7, 'set variant 1 as default variant and switch to variant 2')
        overview.variant.setAsDefault('variant 1');
        overview.variant.switchTo('variant 2')

        cy.step(8, 'remove current variant and check current variant switch to variant 1')
        overview.variant.remove('variant 2');
        overview.variant.checkCurrentVariant();

        cy.step(9, 'remove current default variant and check variant switch to standard')
        overview.variant.remove('variant 1');
        overview.variant.checkCurrentVariant();
    });

    it.skip('Pin variant', () => {
        cy.total(9);

        cy.step(1, 'open overview page')
        let overview = new OverviewPage();
        overview.visit();

        cy.step(2, 'save variant 1 and pin to home page')
        let home = new HomePage();
        overview.filter.selectDateFilter(FilterLabel.POSTING_DATE, ['01/01/2016...03/31/2016'], true);
        overview.variant.saveAs('variant 1');
        overview.variant.pinTo('tile 1');
        overview.navigateToHome(); 
        
        cy.step(3, 'open pinned variant and check status')
        home.openTile('tile 1');
        overview.checkOverall();

        cy.step(4, 'update pinned variant in page')
        overview.filter.selectEnumFilter(FilterLabel.DOCUMENT_TYPE_DISPLAY_NAME, ['A/R Invoice', 'Sales Order'], true);
        overview.variant.save();

        //TODO: BUG: SBO100-7749
        cy.step(5, 'reopen pinned variant and check status after update')
        overview.navigateToHome();
        home.openTile('tile 1');
        overview.checkOverall();

        //TODO: BUG: SBO100-7744
        cy.step(5, 'remove this variant in page and check status after reopen')
        overview.variant.remove('variant 1');
        overview.navigateToHome();
        home.openTile('tile 1');
        overview.checkOverall();

        cy.step(6, 'clear tile in home page')
        home.removeTile('tile 1');
    });

})
  