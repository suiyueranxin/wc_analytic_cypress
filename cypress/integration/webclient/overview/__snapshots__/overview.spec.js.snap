exports[`Sales Analysis Overview > Open page at first time #0`] =
{
  "DomText": "Standard"
};

exports[`Sales Analysis Overview > Open page at first time #1`] =
{
  "DomText": "Posting Date"
};

exports[`Sales Analysis Overview > Open page at first time #2`] =
{
  "DomText": "Document Type Display Name"
};

exports[`Sales Analysis Overview > Open page at first time #3`] =
{
  "DomText": "Business Partner Code"
};

exports[`Sales Analysis Overview > Open page at first time #4`] =
{
  "DomText": "Business Partner Group Name"
};

exports[`Sales Analysis Overview > Open page at first time #5`] =
{
  "DomText": "Item Code"
};

exports[`Sales Analysis Overview > Open page at first time #6`] =
{
  "DomText": "Item Group"
};

exports[`Sales Analysis Overview > Open page at first time #7`] =
{
  "DomText": "Year:Begin...Today"
};

exports[`Sales Analysis Overview > Open page at first time #8`] =
{
  "DomText": "A/R Invoice"
};

exports[`Sales Analysis Overview > Open page at first time #9`] =
{
  "DomText": "Sales Amount by Year"
};

exports[`Sales Analysis Overview > Open page at first time #10`] =
{
  "DomText": ""
};

exports[`Sales Analysis Overview > Open page at first time #11`] =
{
  "DomText": "0"
};

exports[`Sales Analysis Overview > Change filters of different data types and operators #0`] =
{
  "DomText": "1.3"
};

exports[`Sales Analysis Overview > Change filters of different data types and operators #1`] =
{
  "DomText": "M"
};

exports[`Sales Analysis Overview > Change filters of different data types and operators #2`] =
{
  "DomText": "2.7"
};

exports[`Sales Analysis Overview > Change filters of different data types and operators #3`] =
{
  "DomText": "M"
};

exports[`Sales Analysis Overview > Change filters of different data types and operators #4`] =
{
  "DomText": "1.1"
};

exports[`Sales Analysis Overview > Change filters of different data types and operators #5`] =
{
  "DomText": "M"
};

exports[`Sales Analysis Overview > Change filters of different data types and operators #6`] =
{
  "DomText": "720"
};

exports[`Sales Analysis Overview > Change filters of different data types and operators #7`] =
{
  "DomText": "K"
};

exports[`Sales Analysis Overview > Navigate to chart #0`] =
{
  "DomText": "An Interactive Line Chart "
};

exports[`Sales Analysis Overview > Navigate to chart #1`] =
{
  "DomText": "Net Sales Amount (LC)"
};

exports[`Sales Analysis Overview > Navigate to chart #2`] =
{
  "DomText": "Posting Year"
};

exports[`Sales Analysis Overview > Navigate to chart #3`] =
{
  "DomText": "All"
};

exports[`Sales Analysis Overview > Navigate to chart #4`] =
{
  "DomText": "An Interactive Line Chart "
};

exports[`Sales Analysis Overview > Navigate to chart #5`] =
{
  "DomText": "Net Sales Amount (LC)"
};

exports[`Sales Analysis Overview > Navigate to chart #6`] =
{
  "DomText": "Gross Profit (LC)"
};

exports[`Sales Analysis Overview > Navigate to chart #7`] =
{
  "DomText": "Posting Year and Month"
};

exports[`Sales Analysis Overview > Navigate to chart #8`] =
{
  "DomText": "All"
};

exports[`Sales Analysis Overview > Navigate to chart #9`] =
{
  "DomText": "An Interactive Stacked Column Chart "
};

exports[`Sales Analysis Overview > Navigate to chart #10`] =
{
  "DomText": "Net Sales Amount (LC)"
};

exports[`Sales Analysis Overview > Navigate to chart #11`] =
{
  "DomText": "Posting Year and Quarter"
};

exports[`Sales Analysis Overview > Navigate to chart #12`] =
{
  "DomText": "Business Partner Group Name"
};

exports[`Sales Analysis Overview > Navigate to chart #13`] =
{
  "DomText": "Net Sales Amount (LC)"
};

exports[`Sales Analysis Overview > Navigate to chart #14`] =
{
  "DomText": "Item Description & Code"
};

exports[`Sales Analysis Overview > Navigate to chart #15`] =
{
  "DomText": "All"
};

exports[`Sales Analysis Overview > Navigate to chart #16`] =
{
  "DomText": "An Interactive Donut Chart "
};

exports[`Sales Analysis Overview > Navigate to chart #17`] =
{
  "DomText": "Net Sales Amount (LC)"
};

exports[`Sales Analysis Overview > Navigate to chart #18`] =
{
  "DomText": "Sales Employee or Buyer Name"
};

exports[`Sales Analysis Overview > Navigate to chart #19`] =
{
  "DomText": "An Interactive Heat Map "
};

exports[`Sales Analysis Overview > Navigate to chart #20`] =
{
  "DomText": "Net Sales Amount (LC)"
};

exports[`Sales Analysis Overview > Navigate to chart #21`] =
{
  "DomText": "Posting Year and Month"
};

exports[`Sales Analysis Overview > Navigate to chart #22`] =
{
  "DomText": "Bill-To Country"
};

exports[`Sales Analysis Overview > Save as, update, remove and switch variant #0`] =
{
  "DomText": "AT variant 1"
};

exports[`Sales Analysis Overview > Save as, update, remove and switch variant #1`] =
{
  "DomText": "Posting Date"
};

exports[`Sales Analysis Overview > Save as, update, remove and switch variant #2`] =
{
  "DomText": "Document Type Display Name"
};

exports[`Sales Analysis Overview > Save as, update, remove and switch variant #3`] =
{
  "DomText": "Business Partner Code"
};

exports[`Sales Analysis Overview > Save as, update, remove and switch variant #4`] =
{
  "DomText": "Business Partner Group Name"
};

exports[`Sales Analysis Overview > Save as, update, remove and switch variant #5`] =
{
  "DomText": "Item Code"
};

exports[`Sales Analysis Overview > Save as, update, remove and switch variant #6`] =
{
  "DomText": "Item Group"
};

exports[`Sales Analysis Overview > Save as, update, remove and switch variant #7`] =
{
  "DomText": "Year:Begin...Today"
};

exports[`Sales Analysis Overview > Save as, update, remove and switch variant #8`] =
{
  "DomText": "A/R Invoice"
};

exports[`Sales Analysis Overview > Save as, update, remove and switch variant #9`] =
{
  "DomText": "0"
};

exports[`Sales Analysis Overview > Save as, update, remove and switch variant #10`] =
{
  "DomText": "Standard"
};

exports[`Sales Analysis Overview > Save as, update, remove and switch variant #11`] =
{
  "DomText": "AT variant 1"
};

exports[`Sales Analysis Overview > Save as, update, remove and switch variant #12`] =
{
  "DomText": "0"
};

exports[`Sales Analysis Overview > Save as, update, remove and switch variant #13`] =
{
  "DomText": "AT variant 1"
};

exports[`Sales Analysis Overview > Save as, update, remove and switch variant #14`] =
{
  "DomText": "Sales Amount by Year"
};

exports[`Sales Analysis Overview > Save as, update, remove and switch variant #15`] =
{
  "DomText": ""
};

exports[`Sales Analysis Overview > Save as, update, remove and switch variant #16`] =
{
  "DomText": "0"
};

exports[`Sales Analysis Overview > Set default variant #0`] =
{
  "DomText": "variant 1"
};

exports[`Sales Analysis Overview > Open page at first time #12`] =
{
  "DomText": "Sales and Profit by Month"
};

exports[`Sales Analysis Overview > Open page at first time #13`] =
{
  "DomText": " "
};

exports[`Sales Analysis Overview > Open page at first time #14`] =
{
  "DomText": "Sales Amount by Quarter"
};

exports[`Sales Analysis Overview > Open page at first time #15`] =
{
  "DomText": "By Customer Group"
};

exports[`Sales Analysis Overview > Open page at first time #16`] =
{
  "DomText": "Top 20 Item Sales Amounts"
};

exports[`Sales Analysis Overview > Open page at first time #17`] =
{
  "DomText": " "
};

exports[`Sales Analysis Overview > Open page at first time #18`] =
{
  "DomText": "0 of 0"
};

exports[`Sales Analysis Overview > Open page at first time #19`] =
{
  "DomText": "Sales Amount by Sales Employee"
};

exports[`Sales Analysis Overview > Open page at first time #20`] =
{
  "DomText": " "
};

exports[`Sales Analysis Overview > Open page at first time #21`] =
{
  "DomText": "Country Sales by Month"
};

exports[`Sales Analysis Overview > Open page at first time #22`] =
{
  "DomText": "By Amount"
};

exports[`Sales Analysis Overview > Show details #0`] =
{
  "DomText": "Selected Total: 1,344,100.750"
};

exports[`Sales Analysis Overview > Show details #1`] =
{
  "DomText": "Details (1)"
};

exports[`Sales Analysis Overview > Show details #2`] =
{
  "DomText": "Posting Year2016Net Sales Amount (LC)1,344,100.750"
};

exports[`Sales Analysis Overview > Show details #3`] =
{
  "DomText": ""
};

exports[`Sales Analysis Overview > Show details #4`] =
{
  "DomText": "Details (1)"
};

exports[`Sales Analysis Overview > Show details #5`] =
{
  "DomText": "Posting Year2016Net Sales Amount (LC)1,344,100.750"
};

exports[`Sales Analysis Overview > Show details #6`] =
{
  "DomText": "Selected Total: 551,855.250"
};

exports[`Sales Analysis Overview > Show details #7`] =
{
  "DomText": "Details (2)"
};

exports[`Sales Analysis Overview > Show details #8`] =
{
  "DomText": "Posting Year2016Net Sales Amount (LC)1,344,100.750"
};

exports[`Sales Analysis Overview > Show details #9`] =
{
  "DomText": ""
};

exports[`Sales Analysis Overview > Show details #10`] =
{
  "DomText": "Details (1)"
};

exports[`Sales Analysis Overview > Show details #11`] =
{
  "DomText": "Posting Year2016Net Sales Amount (LC)1,344,100.750"
};

exports[`Sales Analysis Overview > Show details #12`] =
{
  "DomText": ""
};

exports[`Sales Analysis Overview > Show details #13`] =
{
  "DomText": "Details (1)"
};

exports[`Sales Analysis Overview > Show details #14`] =
{
  "DomText": "Posting Year2016Net Sales Amount (LC)1,344,100.750"
};

exports[`Sales Analysis Overview > Show details #15`] =
{
  "DomText": ""
};

exports[`Sales Analysis Overview > Show details #16`] =
{
  "DomText": "Details (1)"
};

exports[`Sales Analysis Overview > Show details #17`] =
{
  "DomText": "Posting Year2016Net Sales Amount (LC)1,344,100.750"
};

exports[`Sales Analysis Overview > Set default variant #1`] =
{
  "DomText": "variant 2"
};

exports[`Sales Analysis Overview > Set default variant #2`] =
{
  "DomText": "Standard"
};

exports[`Sales Analysis Overview > Set default variant #3`] =
{
  "DomText": "Standard"
};

exports[`Sales Analysis Overview > Pin variant #0`] =
{
  "DomText": "variant 1"
};

exports[`Sales Analysis Overview > Pin variant #1`] =
{
  "DomText": "Posting Date"
};

exports[`Sales Analysis Overview > Pin variant #2`] =
{
  "DomText": "Document Type Display Name"
};

exports[`Sales Analysis Overview > Pin variant #3`] =
{
  "DomText": "Business Partner Code"
};

exports[`Sales Analysis Overview > Pin variant #4`] =
{
  "DomText": "Business Partner Group Name"
};

exports[`Sales Analysis Overview > Pin variant #5`] =
{
  "DomText": "Item Code"
};

exports[`Sales Analysis Overview > Pin variant #6`] =
{
  "DomText": "Item Group"
};

exports[`Sales Analysis Overview > Set default variant #4`] =
{
  "DomText": "Standard"
};

exports[`Sales Analysis Overview > Pin variant #7`] =
{
  "DomText": "01/01/2016...03/31/2016"
};

exports[`Sales Analysis Overview > Pin variant #8`] =
{
  "DomText": "A/R Invoice"
};

exports[`Sales Analysis Overview > Pin variant #9`] =
{
  "DomText": "1.3"
};

exports[`Sales Analysis Overview > Pin variant #10`] =
{
  "DomText": "M"
};

exports[`Sales Analysis Overview > Pin variant #11`] =
{
  "DomText": "variant 1"
};

exports[`Sales Analysis Overview > Pin variant #12`] =
{
  "DomText": "Posting Date"
};

exports[`Sales Analysis Overview > Pin variant #13`] =
{
  "DomText": "Document Type Display Name"
};

exports[`Sales Analysis Overview > Pin variant #14`] =
{
  "DomText": "Business Partner Code"
};

exports[`Sales Analysis Overview > Pin variant #15`] =
{
  "DomText": "Business Partner Group Name"
};

exports[`Sales Analysis Overview > Pin variant #16`] =
{
  "DomText": "Item Code"
};

exports[`Sales Analysis Overview > Pin variant #17`] =
{
  "DomText": "Item Group"
};

exports[`Sales Analysis Overview > Pin variant #18`] =
{
  "DomText": "01/01/2016...03/31/2016"
};

exports[`Sales Analysis Overview > Pin variant #19`] =
{
  "DomText": "A/R Invoice"
};

exports[`Sales Analysis Overview > Pin variant #20`] =
{
  "DomText": "1.3"
};

exports[`Sales Analysis Overview > Pin variant #21`] =
{
  "DomText": "M"
};

exports[`Sales Analysis Overview > Pin variant #22`] =
{
  "DomText": "Standard"
};

exports[`Sales Analysis Overview > Pin variant #23`] =
{
  "DomText": "Posting Date"
};

exports[`Sales Analysis Overview > Pin variant #24`] =
{
  "DomText": "Document Type Display Name"
};

exports[`Sales Analysis Overview > Pin variant #25`] =
{
  "DomText": "Business Partner Code"
};

exports[`Sales Analysis Overview > Pin variant #26`] =
{
  "DomText": "Business Partner Group Name"
};

exports[`Sales Analysis Overview > Pin variant #27`] =
{
  "DomText": "Item Code"
};

exports[`Sales Analysis Overview > Pin variant #28`] =
{
  "DomText": "Item Group"
};

exports[`Sales Analysis Overview > Pin variant #29`] =
{
  "DomText": "01/01/2016...03/31/2016"
};

exports[`Sales Analysis Overview > Pin variant #30`] =
{
  "DomText": "A/R Invoice"
};

exports[`Sales Analysis Overview > Pin variant #31`] =
{
  "DomText": "1.3"
};

exports[`Sales Analysis Overview > Pin variant #32`] =
{
  "DomText": "M"
};
